src/rates/currencies.ts

```diff
@@ -16,19 +16,25 @@ class Currency {
     this._fiat = fiat;
     this._sources = sources;
 
-    for (const param of this._sources) {
-      /**
-       * Using croner instead of @nestjs/schedule
-       * because of issue: https://github.com/kelektiv/node-cron/issues/805
-       */
-      new Cron(param.pattern, () => {
+    // Senior-level: Clean, clear, and DRY cron scheduling
+    const totalCurrencies = this._sources.length;
+    const baseMinuteOffsets = Array.from({ length: totalCurrencies }, (_, i) => i);
+    const intervals = Array.from({ length: totalCurrencies }, (_, i) => 10 + (i % 9)); // 10-18 min
+
+    this._sources.forEach((param, idx) => {
+      const second = Math.floor(Math.random() * 60);
+      const minuteOffset = baseMinuteOffsets[idx];
+      const interval = intervals[idx];
+      const minutePattern = Array.from({ length: Math.floor(60 / interval) }, (_, i) => (i * interval + minuteOffset) % 60).join(",");
+      const cronPattern = `${second} ${minutePattern} * * * *`;
+      new Cron(cronPattern, () => {
```

Comment:
even though this implementation is brilliant, i'd rather we stick to the former implementation because this config below `{ source: new Binance(), pattern: CronExpression.EVERY_5_MINUTES }` can be altered for difference currencies

e.g IDR could be EVERY_10_MINUTES and NGN could be EVERY_5_MINUTES

they could be tweaked based on the volatility of the currency to usd

would be great set this config independently instead of dynamically setting them according to a mathematical pattern.

Reply:
gm! Thanks for the review :D
Your reasoning makes sense, sorry for going overboard 😅 
Reverting this change
Pushed 🫡